middleware/instrument.go

```diff
@@ -68,9 +68,20 @@ func (i Instrument) Wrap(next http.Handler) http.Handler {
 			next.ServeHTTP(ww, r)
 		})
 
-		i.Duration.WithLabelValues(r.Method, route, strconv.Itoa(respMetrics.Code), isWS).Observe(respMetrics.Duration.Seconds())
 		i.RequestBodySize.WithLabelValues(r.Method, route).Observe(float64(rBody.read))
 		i.ResponseBodySize.WithLabelValues(r.Method, route).Observe(float64(respMetrics.Written))
+
+		histogram := i.Duration.WithLabelValues(r.Method, route, strconv.Itoa(respMetrics.Code), isWS)
+		if traceID, ok := ExtractSampledTraceID(r.Context()); ok {
+			// Need to type-convert the Observer to an
+			// ExemplarObserver. This will always work for a
+			// HistogramVec.
+			histogram.(prometheus.ExemplarObserver).ObserveWithExemplar(
+				respMetrics.Duration.Seconds(), prometheus.Labels{"traceID": traceID},
```

Comment:
Is it OK to hard-code `"traceID"` ?   I didn't find the documentation on `ObserveWithExemplar()` very helpful in understanding what those labels are used for.

Reply:
From the perspective of prometheus/client_golang, these are completely free-form, and it's totally up to the user what to use them for. We are using them for trace IDs, but others might want to point to a log entry. Or they might want to put in an additional parameter for particular events, e.g. the query string for a query that resulted in an exceptionally long processing time. We are still at the beginning of the exemplar journey, so we have to see what they will be used for.

Reply:
So why "traceID" here in particular? That's just the convention that we came up with. As we are the first doing it in the Prometheus space, we have the power of naming things… Despite hardcoding it here, it might still change. People will come up with various bikeshedding arguments (e.g. it's more common in Prometheus land to use snake case, so some might argue it should be `trace_id`), and if you prefer, we could make it configurable here.

Reply:
OK let's move forward and change it later if we need to.